api/csi.go

```diff
@@ -206,6 +228,28 @@ type CSIMountOptions struct {
 // API or in Nomad's logs.
 type CSISecrets map[string]string
 
+func (q *QueryOptions) SetHeadersFromCSISecrets(secrets CSISecrets) {
+	pairs := []string{}
+	for k, v := range secrets {
+		pairs = append(pairs, fmt.Sprintf("%v=%v", k, v))
+	}
+	if q.Headers == nil {
+		q.Headers = map[string]string{}
+	}
+	q.Headers["X-Nomad-CSI-Secrets"] = strings.Join(pairs, ",")
+}
+
+func (q *WriteOptions) SetHeadersFromCSISecrets(secrets CSISecrets) {
```

Comment:
```suggestion
func (w *WriteOptions) SetHeadersFromCSISecrets(secrets CSISecrets) {
```
nit-picking, feel free to ignore.